aten/src/ATen/native/cuda/Indexing.cu

```diff
@@ -900,10 +900,10 @@ void index_add_cuda_impl(const Tensor& self, int64_t dim, const Tensor& index, c
   // total size of the tensor ignoring dimension `dim`;
   // -the number of index we are choosing, which is the total size
   // of the tensor `index`.
-  const ptrdiff_t sliceSize = getSliceSize(self_, dim, index, source_);
-  const ptrdiff_t sourceTotalSize = source.numel();
+  const size_t sliceSize = getSliceSize(self_, dim, index, source_);
+  const long sourceTotalSize = source.numel();
```

Comment:
Avoid long, for consistency?  Other calls to numel store into int64_t.
```suggestion
  const int64_t sourceTotalSize = source.numel();
```

Reply:
> Avoid long, for consistency? Other calls to numel store into int64_t.
This file has many types, ptrdiff_t (not sure why use this point diff type to store sizes of tensors or number of elements of tensors), long (The signature of `numel` returns `long` atm), and int64_t. 
Use `int64_t` is ok and unsigned 64 bit integer `uint64_t` is even better as I don't think we expect the number of elements or size of tensors to be negative integers.